packages/interchain-token-service-api/src/primitives.rs

```diff
@@ -0,0 +1,90 @@
+use crate::error::Error;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{HexBinary, Uint256};
+use error_stack::Report;
+
+#[cw_serde]
+#[derive(Eq)]
+pub struct TokenId {
+    #[serde(with = "axelar_wasm_std::hex")]
+    #[schemars(with = "String")]
+    pub id: [u8; 32],
+}
+
+#[cw_serde]
+#[derive(Eq, Copy)]
+#[repr(u8)]
+pub enum TokenManagerType {
+    NativeInterchainToken = 0,
+    MintBurnFrom = 1,
+    LockUnlock = 2,
+    LockUnlockFee = 3,
+    MintBurn = 4,
+    Gateway = 5,
+}
+
+#[cw_serde]
+#[derive(Eq)]
+pub enum ITSMessage {
+    InterchainTransfer {
+        token_id: TokenId,
+        source_address: HexBinary,
+        destination_address: HexBinary,
```

Comment:
Should probably use the `Address` type that we use for `Message`. It's defined in `router_api::primitives`. Will make composability a bit cleaner.